modules/katamari/src/main/ts/ephox/katamari/api/Type.ts

```diff
@@ -1,35 +1,43 @@
-const typeOf = function (x: any) {
-  if (x === null) {
-    return 'null';
-  }
-  const t = typeof x;
-  if (t === 'object' && (Array.prototype.isPrototypeOf(x) || x.constructor && x.constructor.name === 'Array')) {
-    return 'array';
-  }
-  if (t === 'object' && (String.prototype.isPrototypeOf(x) || x.constructor && x.constructor.name === 'String')) {
-    return 'string';
-  }
-  return t;
-};
+const typeOf = (x: any): string =>
+  x === null ? 'null' : isArray(x) ? 'array' : isString(x) ? 'string' : typeof x;
```

Comment:
Not sure nested ternaries is easier to read than just an if statement here.

Reply:
Agreed and we don't allow nested ternaries normally.

Reply:
Reads fine to me. 

The point of a ternary is that it's an expression, not a statement.

Reply:
Nevertheless, I'll follow your coding standard. Fixed.

Reply:
I do prefer the if/else style over what we returned to, even if we keep the logic, because a minification tool will translate it into a ternary.

Reply:
Yep. Moving back to this style reduced the output size.